ts/private/ts_lib.bzl

```diff
@@ -12,7 +12,7 @@ ValidOptionsInfo = provider(
 
 # Targets in deps must provide one or the other of these
 DEPS_PROVIDERS = [
-    [DeclarationInfo],
```

Comment:
https://en.wikipedia.org/wiki/Robustness_principle suggests that we should consider keeping DeclarationInfo here as an option. I think it's pretty trivial in the implementation to check it if JsInfo isn't present on a dep

Reply:
One problem with that is that we'd be advertising interop when we wouldn't be picking up the transitive npm dependencies from any ts_projects the produce DeclarationInfo

Reply:
yes that will be confusing for users that some deps have different behavior. I'm okay with this like it is